pcdsdevices/utils.py

```diff
@@ -156,3 +158,47 @@ def get_component(obj):
         return None
 
     return getattr(type(obj.parent), obj.attr_name, None)
+
+
+def schedule_task(func, args=None, kwargs=None, delay=None):
+    """
+    Use ophyd's dispatcher to schedule a task for later.
+
+    This is basically the function I was hoping to find in ophyd.
+    Schedules a task for the utility thread if we're in some arbitrary thread,
+    schedules a task for the same thread if we're in one of ophyd's callback
+    queues already.
+    """
+    if args is None:
+        args = ()
+    if kwargs is None:
+        kwargs = {}
+
+    dispatcher = ophyd.cl.get_dispatcher()
+
+    # Check if we're already in an ophyd dispatcher thread
+    current_thread = threading.currentThread()
+    matched_thread = None
+    for name, thread in dispatcher.threads.items():
+        if thread == current_thread:
+            matched_thread = name
+            context = dispatcher.get_thread_context(matched_thread)
+            break
+
+    if delay is None:
+        if matched_thread is None:
```

Comment:
Very sensible. I hadn't considered doing it this way, but I like it. 👍 

To reduce code duplication, you might consider making this a closure:
```python
def schedule():
  if matched_thread is None:
    schedule...
  else:
    context.event_thread...

if delay is None:
     schedule()
else:
    Timer(... schedule...).start()
```

Reply:
Yes, that is cleaner. I will rewrite it like this and then hopefully deploy the darn thing